src/editor/reducers/file-source-inputs-reducer.js

```diff
@@ -0,0 +1,32 @@
+function produceFileSourceInputs(state) {
```

Comment:
it was requested to add a reducer specifically for the inputs for the `AddFileSource` component state. I am sure there will be further feedback on this PR that will necessitate more changes, so I will get to this before that next update or during it.